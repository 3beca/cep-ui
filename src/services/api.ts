import {
    fetchApi,
    APIRequestInfo,
    APIResponseData,
    APIError,
    isAPIError
} from '../utils/fetch-api';

export type ServiceList<T> = {
    results: T[];
    next?: string | undefined;
    prev?: string | undefined;
};
export type ServiceError = {
    statusCode: number,
    error: string,
    message: string
};
export type Entity = {
    id: string;
};
export type WithDate = {
    createdAt: string;
    updatedAt: string;
};
export type ServiceDeleted = {
    state: 'DELETED'|'REJECTED';
    error?: ServiceError;
} & Entity;
export type EventType = {
    name: string;
    url: string;
} & Entity & WithDate;
export type EventTypeList = ServiceList<EventType>;
export type EventTypeError = ServiceError;
export type EventTypeDeleted = ServiceDeleted;

export type Target = {
    name: string;
    url: string;
} & Entity & WithDate;
export type TargetList = ServiceList<Target>;
export type TargetError = ServiceError;
export type TargetDeleted = ServiceDeleted;

export type RuleTypes = 'sliding'|'hopping'|'tumbling'|'none';
export type Rule = {
    name: string;
    type: RuleTypes;
    targetId: string;
    eventTypeId: string;
    skipOnConsecutivesMatches?: boolean;
    filters: any;
} & Entity & WithDate;
export type RuleList = ServiceList<Rule>;
export type RuleError = ServiceError;
export type RuleDeleted = ServiceDeleted;

export type EventLog = {
    eventTypeId: string;
    eventTypeName: string;
    payload: {[key: string]: any;};
    requestId: string;
    createdAt: string;
    rules: {
        id: string;
        name: string;
        targetId: string;
    }[];
    targets: {
        id: string;
        name: string;
        response: {
            statusCode: number;
            body: {[key: string]: any;};
        };
    }[];
} & Entity;

export const getListRequest = (baseURL: string, config: APIRequestInfo) => async <T>(path: string, page: number = 1, size: number = 10, filter: string = ''): Promise<APIResponseData<ServiceList<T>>|APIError<ServiceError>> => {
    const url = `${baseURL}${path}/?page=${page}&pageSize=${size}${filter ? `&search=${filter}` : ''}`;
    return fetchApi<undefined, ServiceList<T>, ServiceError>(url, {...config, method: 'GET'});
};
export const deleteRequest = (baseURL: string, config: APIRequestInfo) => async (path: string, eventTypeIds: string|string[]): Promise<(APIResponseData<ServiceDeleted[]>|APIError<ServiceError>)> => {
    const eventTypesArray = ((Array.isArray(eventTypeIds)) ? eventTypeIds : [eventTypeIds]).filter(id => !!id);

    if(!eventTypeIds || eventTypesArray.length < 1) {
        return {
            errorCode: 500,
            errorMessage: '',
            error: {
                statusCode: 500,
                error: 'Missing id',
                message: 'id is an invalid id value or array'
            }
        };
    }
    const requests = [];
    for (const eventTypeId of eventTypesArray) {
        const url = `${baseURL}${path}/${eventTypeId}`;
        requests.push(fetchApi<undefined, undefined, ServiceError>(url, {...config, method: 'DELETE'}));
    }
    const responses = await Promise.all(requests);
    const data: ServiceDeleted[] = responses.map((response, idx) => {
        const deletedResponse: ServiceDeleted = {
            id: eventTypesArray[idx],
            state: 'DELETED'
        };
        if(isAPIError(response)) {
            deletedResponse.state = 'REJECTED';
            deletedResponse.error = response.error;
        }
        return deletedResponse;
    });
    return {
        status: 200,
        data
    };
};
export type Api = {
    getListRequest<T>(path: string, page?: number, size?: number, filter?: string): Promise<APIResponseData<ServiceList<T>>|APIError<ServiceError>>;
    deleteRequest(path: string, ids: string|string[]): Promise<(APIResponseData<ServiceDeleted[]>|APIError<ServiceError>)>;
};
export const buildApiService = (server: string): Api => {
    const config: APIRequestInfo = {
        method: 'GET',
        headers: {}
    };
    return {
        getListRequest: getListRequest(server, config),
        deleteRequest: deleteRequest(server, config)
    };
};
