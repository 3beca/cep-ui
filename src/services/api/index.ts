import {
    fetchApi,
    APIRequestInfo,
    APIResponseData,
    APIError,
    isAPIError,
    APIBody
} from '../../utils/fetch-api';
import {
    ServiceList,
    ServiceError,
    ServiceDeleted,
} from './models';
export * from './models';

export const getListRequest = (baseURL: string, config: APIRequestInfo) => async <T>(path: string, page: number = 1, size: number = 10, filter: string = ''): Promise<APIResponseData<ServiceList<T>>|APIError<ServiceError>> => {
    const url = `${baseURL}${path}/?page=${page}&pageSize=${size}${filter ? `&search=${filter}` : ''}`;
    return fetchApi<undefined, ServiceList<T>, ServiceError>(url, {...config, method: 'GET'});
};
export const deleteRequest = (baseURL: string, config: APIRequestInfo) => async (path: string, entityIds: string|string[]): Promise<(APIResponseData<ServiceDeleted[]>|APIError<ServiceError>)> => {
    const entityArray = ((Array.isArray(entityIds)) ? entityIds : [entityIds]).filter(id => !!id);

    if(!entityIds || entityArray.length < 1) {
        return {
            errorCode: 500,
            errorMessage: '',
            error: {
                statusCode: 500,
                error: 'Missing id',
                message: 'id is an invalid id value or array'
            }
        };
    }
    const requests = [];
    for (const entityId of entityArray) {
        const url = `${baseURL}${path}/${entityId}`;
        requests.push(fetchApi<undefined, undefined, ServiceError>(url, {...config, method: 'DELETE'}));
    }
    const responses = await Promise.all(requests);
    const data: ServiceDeleted[] = responses.map((response, idx) => {
        const deletedResponse: ServiceDeleted = {
            id: entityArray[idx],
            state: 'DELETED'
        };
        if(isAPIError(response)) {
            deletedResponse.state = 'REJECTED';
            deletedResponse.error = response.error;
        }
        return deletedResponse;
    });
    return {
        status: 200,
        data
    };
};
export const createRequest = (baseURL: string, config: APIRequestInfo) => async <T extends APIBody>(path: string, body: Partial<T>): Promise<APIResponseData<T>|APIError<ServiceError>> => {
    const url = `${baseURL}${path}`;
    return fetchApi<Partial<T>, T, ServiceError>(url, {...config, method: 'POST', headers: {...config.headers, 'content-type': 'application/json'}, body});
};
export type Api = {
    getListRequest<T>(path: string, page?: number, size?: number, filter?: string): Promise<APIResponseData<ServiceList<T>>|APIError<ServiceError>>;
    deleteRequest(path: string, ids: string|string[]): Promise<(APIResponseData<ServiceDeleted[]>|APIError<ServiceError>)>;
    createRequest<T extends APIBody>(path: string, body: Partial<T>): Promise<APIResponseData<T>|APIError<ServiceError>>;
};
export const buildApiService = (server: string): Api => {
    const config: APIRequestInfo = {
        method: 'GET',
        headers: {}
    };
    return {
        getListRequest: getListRequest(server, config),
        deleteRequest: deleteRequest(server, config),
        createRequest: createRequest(server, config)
    };
};
