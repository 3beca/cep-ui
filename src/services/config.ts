export const BASE_URL = window._env_.BASE_URL;
export const EVENT_TYPE_URL = '/admin/event-types';