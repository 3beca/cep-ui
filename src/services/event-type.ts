import {
    fetchApi,
    APIRequestInfo,
    APIResponseData,
    APIError,
    isAPIError
} from '../fetch-api';
import {
    BASE_URL,
    EVENT_TYPE_URL
} from './config';

export type EventType = {
    name: string;
    id: string;
    url: string;
    createdAt: string;
    updatedAt: string;
};
export type EventTypeList = {
    results: EventType[];
    next?: string | undefined;
    prev?: string | undefined;
};
export type EventTypeError = {
    statusCode: number,
    error: string,
    message: string
};
export type EventTypeDeleted = {
    id: string;
    state: 'DELETED'|'REJECTED';
    error?: EventTypeError;
};
export const getEventTypeList = (baseURL: string, config: APIRequestInfo) => async (page: number = 1, size: number = 10): Promise<APIResponseData<EventTypeList>|APIError<EventTypeError>> => {
    const url = `${baseURL}${EVENT_TYPE_URL}/?page=${page}&pageSize=${size}`;
    return fetchApi<undefined, EventTypeList, EventTypeError>(url, {...config, method: 'GET'});
};
export const deleteEventType = (baseURL: string, config: APIRequestInfo) => async (eventTypeIds: string|string[]): Promise<(APIResponseData<EventTypeDeleted[]>|APIError<EventTypeError>)> => {
    const eventTypesArray = (Array.isArray(eventTypeIds)) ? eventTypeIds : [eventTypeIds];
    if(!eventTypeIds || eventTypesArray.length < 1) {
        return {
            errorCode: 500,
            errorMessage: '',
            error: {
                statusCode: 500,
                error: 'Missing EventType id',
                message: 'eventTypeIds is an invalid EventTypeId value or array'
            }
        };
    }
    const requests = [];
    for (const eventTypeId of eventTypesArray) {
        const url = `${baseURL}${EVENT_TYPE_URL}/${eventTypeId}`;
        requests.push(fetchApi<undefined, undefined, EventTypeError>(url, {...config, method: 'DELETE'}));
    }
    const responses = await Promise.all(requests);
    const data: EventTypeDeleted[] = responses.map((response, idx) => {
        const deletedResponse: EventTypeDeleted = {
            id: eventTypesArray[idx],
            state: 'DELETED'
        };
        if(isAPIError(response)) {
            deletedResponse.state = 'REJECTED';
            deletedResponse.error = response.error;
        }
        return deletedResponse;
    });
    return {
        status: 200,
        data
    };
};
export type EventTypeApi = {
    getEventTypeList(page?: number, size?: number): Promise<APIResponseData<EventTypeList>|APIError<EventTypeError>>;
    deleteEventType(eventTypeIds: string|string[]): Promise<(APIResponseData<EventTypeDeleted[]>|APIError<EventTypeError>)>;
};
export const EventTypeApi = (): EventTypeApi => {
    const config: APIRequestInfo = {
        method: 'GET',
        headers: {}
    };
    return {
        getEventTypeList: getEventTypeList(BASE_URL, config),
        deleteEventType: deleteEventType(BASE_URL, config)
    };
};
