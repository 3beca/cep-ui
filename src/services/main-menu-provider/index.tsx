import * as React from 'react';
import { NOOP } from '../../utils';
export type MainMenuToggleContextValue = {
    toggle: () => void;
    close: () => void;
    open: () => void;
};
const MainMenuStateContext = React.createContext(false);
const MainMenuToggleContext = React.createContext<MainMenuToggleContextValue>({
    toggle: NOOP,
    close: NOOP,
    open: NOOP
});

export const MainMenuProvider: React.FC<{}> = props => {
    const [isOpen, setOpen] = React.useState(false);
    const toggle = React.useCallback(() => setOpen(isOpen => !isOpen), []);
    const open = React.useCallback(() => setOpen(true), []);
    const close = React.useCallback(() => setOpen(false), []);
    const utils = React.useMemo(
        () => ({
            toggle,
            open,
            close
        }),
        [toggle, open, close]
    );

    return (
        <MainMenuToggleContext.Provider value={utils}>
            <MainMenuStateContext.Provider value={isOpen} {...props} />
        </MainMenuToggleContext.Provider>
    );
};

export const useMainMenuState = () => {
    return React.useContext(MainMenuStateContext);
};

export const useMainMenuToggle = () => {
    return React.useContext(MainMenuToggleContext);
};
