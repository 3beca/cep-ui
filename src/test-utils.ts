import nock from 'nock';

export const setupNock = (url: string) => {
    const server = nock(url).defaultReplyHeaders({ 'access-control-allow-origin': '*' });
    // Skip Preflight CORS OPTION request
    nock(url).intercept(/./, 'OPTIONS').reply(200, undefined, { 'access-control-allow-origin': '*' }).persist();
    return server;
}