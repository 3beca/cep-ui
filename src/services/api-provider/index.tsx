import * as React from 'react';
import { APIRequestInfo, isAPIError } from '../../utils/fetch-api';
import { buildApiService } from '../api';
import { VersionInfo } from '../api/models';
import { BASE_URL, VERSION_URL } from '../config';
import { loadApikey, saveApikey, clearApikey } from '../../utils';
import {
    APIContextState,
    ApiContextActions,
    ValidationState,
    initialContext,
    APIContext,
    UpdateAPIContext
} from './api-context';

// config = {method: 'GET', headers: {'authorization': 'apiKey ' + apiKey}};
export const apiReducer = (
    state: APIContextState,
    action: ApiContextActions
) => {
    switch (action.type) {
        case 'VALIDATING': {
            return {
                isValidating: true,
                isValidated: ValidationState.PENDING,
                isValid: false
            };
        }
        case 'SERVER_NOT_FOUND': {
            return {
                isValidating: false,
                isValidated: ValidationState.NOT_FOUND,
                isValid: false,
                invalidReason: `${action.url} not found`
            };
        }
        case 'NO_REQUIRE_APIKEY': {
            return {
                isValidating: false,
                isValidated: ValidationState.VALIDATED,
                isValid: true,
                requireKey: false,
                api: buildApiService(BASE_URL),
                version: action.version
            };
        }
        case 'REQUIRE_APIKEY': {
            return {
                isValidating: false,
                isValidated: ValidationState.VALIDATED,
                isValid: false,
                requireKey: true,
                invalidReason: action.reason
            };
        }
        case 'SET_APIKEY': {
            const apiKey = action.apiKey;
            const version = action.version;
            const apiInfo: APIRequestInfo = {
                headers: { Authorization: 'apiKey ' + apiKey },
                method: 'GET'
            };
            return {
                isValidating: false,
                isValidated: ValidationState.VALIDATED,
                isValid: true,
                requireKey: true,
                apiKey,
                api: buildApiService(BASE_URL, apiInfo),
                version
            };
        }
        default:
            return state;
    }
};
export const APIProvider: React.FC<{}> = props => {
    const [state, dispatch] = React.useReducer(apiReducer, initialContext);

    const checkApikey = React.useCallback(async (apiKey: string) => {
        dispatch({ type: 'VALIDATING' });
        const apiInfo: APIRequestInfo = {
            headers: { Authorization: 'apiKey ' + apiKey },
            method: 'GET'
        };
        const api = buildApiService(BASE_URL, apiInfo);
        const responseWithtoken = await api.getRequest<VersionInfo>(
            VERSION_URL
        );
        if (isAPIError(responseWithtoken)) {
            clearApikey();
            dispatch({
                type: 'REQUIRE_APIKEY',
                reason: `ApiKey ${apiKey} is NOT valid`
            });
            return;
        } else {
            saveApikey(apiKey);
            dispatch({
                type: 'SET_APIKEY',
                apiKey,
                version: responseWithtoken.data.version
            });
            return;
        }
    }, []);

    const fireValidation = React.useCallback(async () => {
        dispatch({ type: 'VALIDATING' });
        let api = buildApiService(BASE_URL);
        const response = await api.getRequest<VersionInfo>(VERSION_URL);
        if (isAPIError(response)) {
            if (response.errorCode === 500) {
                dispatch({ type: 'SERVER_NOT_FOUND', url: BASE_URL });
                return;
            }
            const apiKey = loadApikey();
            if (!apiKey) {
                dispatch({
                    type: 'REQUIRE_APIKEY',
                    reason: 'apiKey not found'
                });
                return;
            }
            checkApikey(apiKey);
        } else {
            dispatch({
                type: 'NO_REQUIRE_APIKEY',
                version: response.data.version
            });
        }
    }, [checkApikey]);

    const invalidateApiKey = React.useCallback(() => {
        clearApikey();
        fireValidation();
    }, [fireValidation]);

    const setApiKey = React.useCallback(
        (apiKey: string) => {
            checkApikey(apiKey);
        },
        [checkApikey]
    );

    const utils = React.useMemo(
        () => ({
            invalidateApiKey,
            setApiKey
        }),
        [invalidateApiKey, setApiKey]
    );

    React.useEffect(() => {
        fireValidation();
    }, [fireValidation]);

    return (
        <APIContext.Provider value={state}>
            <UpdateAPIContext.Provider value={utils} {...props} />
        </APIContext.Provider>
    );
};

export const useUpdateAPIProvider = () => {
    return React.useContext(UpdateAPIContext);
};

export const useAPIProviderStatus = () => {
    const api = React.useContext(APIContext);
    return {
        showLoading:
            api.isValidating || api.isValidated === ValidationState.PENDING,
        showNoService:
            !api.isValidating && api.isValidated === ValidationState.NOT_FOUND,
        showLogin:
            !api.isValidating &&
            api.requireKey &&
            api.isValidated === ValidationState.VALIDATED &&
            !api.isValid,
        requireApikey: api.requireKey,
        invalidReason: api.invalidReason,
        apiKey: api.apiKey,
        version: api.version
    };
};

export const useAPIProvider = () => {
    const api = React.useContext(APIContext);
    if (!api.api) throw Error('no API available');
    return {
        api: api.api,
        apiKey: api.apiKey,
        version: api.version
    };
};
