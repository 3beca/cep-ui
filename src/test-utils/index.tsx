import * as React from 'react';
import nock from 'nock';
import {Router} from 'react-router-dom';
import {createMemoryHistory, History} from 'history';
import {render} from '@testing-library/react';
import {MainMenuProvider} from '../services/main-menu-provider';
import { EventTypeList, EventTypeError } from '../services/event-type';
import { EVENT_TYPE_URL } from '../services/config';
export * from '@testing-library/react';

type OptionsRenderRouter = {
    route?: string;
    history?: History
};
export function renderInsideApp(
    ui: React.ReactElement,
    {
        route = '/',
        history = createMemoryHistory({initialEntries: [route]}),
        ...renderOptions
    }: OptionsRenderRouter = {}
) {
    const Wrapper: React.FC<{}> = function Wrapper(props) {
        return (
            <Router history={history}>
                <MainMenuProvider {...props} />
            </Router>
        );
    }
    return {
        ...render(ui, {wrapper: Wrapper, ...renderOptions}),
        history
    };
}

export const setupNock = (url: string) => {
    const server = nock(url).defaultReplyHeaders({ 'access-control-allow-origin': '*' });
    // Skip Preflight CORS OPTION request
    nock(url).intercept(/./, 'OPTIONS').reply(200, undefined, { 'access-control-allow-origin': '*' }).persist();
    return server;
}

export const generateEventTypeListWith = (many: number = 5, next = false, prev = false) => {
    const list: EventTypeList = {
        results: Array.from({length: many},
            (_, idx) => ({
                id: idx + '',
                name: 'Elemento ' + idx,
                url: 'http://cep/elemento' + idx,
                createdAt: '2020-01-01T10:10:10.123Z',
                updatedAt: '2020-01-01T10:10:10.123Z'
            })
        )
    };
    if (prev) list.prev = 'http://cep/?page=prev';
    if (next) list.next = 'http://cep/?page=next';
    return list;
};

export const serverGetEventTypeList = (server: nock.Scope, page: number = 1, size: number = 10, status: number = 200, response: EventTypeList|EventTypeError = generateEventTypeListWith(10, false, false)) => {
    return server.get(EVENT_TYPE_URL + `/?page=${page}&pageSize=${size}`).reply(status, response);
};
export const serverDeleteEventType = (server: nock.Scope, eventTypeId: string, status: number = 204, response: undefined|EventTypeError = undefined) => {
    return server.delete(EVENT_TYPE_URL + `/${eventTypeId}`).reply(status, response);
};