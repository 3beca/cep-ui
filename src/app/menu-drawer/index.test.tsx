import * as React from 'react';
import { renderInsideApp, fireEvent } from '../../test-utils';
import MenuDrawer from './index';
import { useMainMenuToggle, useMainMenuState } from '../../services/main-menu-provider';

const mockMainMenuStatus = (useMainMenuState as unknown) as jest.Mock;
const mockClose = (useMainMenuToggle().close as unknown) as jest.Mock;
jest.mock('../../services/main-menu-provider', () => {
    const close = jest.fn();
    return {
        useMainMenuToggle: () => ({
            close
        }),
        useMainMenuState: jest.fn(),
        MainMenuProvider: (props: any) => <div {...props} />
    };
});

afterEach(() => {
    mockMainMenuStatus.mockClear();
    mockClose.mockClear();
});

export const renderMenuOpenedInRoute = (route = '/') => {
    mockMainMenuStatus.mockReturnValueOnce(true);
    const utils = renderInsideApp(<MenuDrawer />, { route });
    utils.getByLabelText(/drawer menu/i);
    const menuEventTypes = utils.getByLabelText(/menu event type list page/);
    const menuTargets = utils.getByLabelText(/menu target list page/);
    const menuRules = utils.getByLabelText(/menu rules list page/);
    const menuEvents = utils.getByLabelText(/^menu event logs list page$/i);

    return {
        ...utils,
        menuEventTypes,
        menuTargets,
        menuRules,
        menuEvents
    };
};

test('MenuDrawer should not render menu when is closed', () => {
    const { queryByLabelText, rerender } = renderInsideApp(<MenuDrawer />);
    expect(queryByLabelText(/drawer menu/i)).not.toBeInTheDocument();
    mockMainMenuStatus.mockReturnValueOnce(true);
    rerender(<MenuDrawer />);
    expect(queryByLabelText(/drawer menu/i)).toBeInTheDocument();
});

test('MenuDrawer should render menu when open in / with Rules selected and snapshot', () => {
    const { container, history } = renderMenuOpenedInRoute();
    expect(history.location.pathname).toEqual('/');
    expect(container).toMatchSnapshot();
});

test('MenuDrawer should render menu when open in / and navigate to targets, snapshot and back', () => {
    const { container, history, menuTargets, menuRules, queryByLabelText } = renderMenuOpenedInRoute();
    mockMainMenuStatus.mockReturnValueOnce(true);
    fireEvent.click(menuTargets);
    expect(history.location.pathname).toEqual('/targets');
    expect(mockClose).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();

    mockMainMenuStatus.mockReturnValueOnce(false);
    fireEvent.click(menuRules);
    expect(history.location.pathname).toEqual('/');
    expect(mockClose).toHaveBeenCalledTimes(2);
    expect(queryByLabelText(/drawer menu/i)).not.toBeInTheDocument();
});

test('MenuDrawer should render menu when open in / and navigate to Event Types, snapshot and back', () => {
    const { container, history, menuRules, menuEventTypes, queryByLabelText } = renderMenuOpenedInRoute();
    mockMainMenuStatus.mockReturnValueOnce(true);
    fireEvent.click(menuEventTypes);
    expect(history.location.pathname).toEqual('/event-types');
    expect(mockClose).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();

    mockMainMenuStatus.mockReturnValueOnce(false);
    fireEvent.click(menuRules);
    expect(history.location.pathname).toEqual('/');
    expect(mockClose).toHaveBeenCalledTimes(2);
    expect(queryByLabelText(/drawer menu/i)).not.toBeInTheDocument();
});

test('MenuDrawer should render menu when open in / and navigate to Events, snapshot and back', () => {
    const { container, history, menuRules, menuEvents, queryByLabelText } = renderMenuOpenedInRoute();
    mockMainMenuStatus.mockReturnValueOnce(true);
    fireEvent.click(menuEvents);
    expect(history.location.pathname).toEqual('/event-logs');
    expect(mockClose).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();

    mockMainMenuStatus.mockReturnValueOnce(false);
    fireEvent.click(menuRules);
    expect(history.location.pathname).toEqual('/');
    expect(mockClose).toHaveBeenCalledTimes(2);
    expect(queryByLabelText(/drawer menu/i)).not.toBeInTheDocument();
});
