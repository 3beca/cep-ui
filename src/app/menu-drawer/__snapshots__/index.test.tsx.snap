// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuDrawer should render menu when open in / and navigate to targets, snapshot and back 1`] = `
<div
  aria-hidden="true"
>
  <div />
</div>
`;

exports[`MenuDrawer should render menu when open in / with EventTypes selected and snapshot 1`] = `
<div
  aria-hidden="true"
>
  <div />
</div>
`;
